compiler/passes/convert-typed-uast.cpp

```diff
@@ -2221,129 +2838,781 @@ void TConverter::setVariableType(const uast::VarLikeDecl* v,
-    if (qt.isType() && !typeExistsAtRuntime(qt.type())) {
-      // a 'type' actual can be left out at this point
-    } else if (qt.isParam()) {
-      // a 'param' actual can be left out at this point
+  Expr* ret = nullptr;
+
+  // Consider the return type of the called function.
+  // Do we need to emit this call at all?
+  // TODO: If the call is producing a type, do we really need to bother?
+  if (!functionExistsAtRuntime(rf)) {
+    // don't emit the call, but instead emit a SymExpr for the relevant
+    // type or param.
+    const auto& qt = rf->returnType();
+    if (qt.isType()) {
+      Type* t = convertType(qt.type());
+      ret = new SymExpr(t->symbol);
     } else {
-      if (fa.hasDefault()) {
-        CHPL_UNIMPL("default arguments");
-      } else {
-        // it's something that exists at runtime, so include it
-        Expr* actualExpr = convertExpr(actualAsts[fa.actualIdx()], rv);
-        c->insertAtTail(actualExpr);
-      }
-      // TODO handle associated actions for 'in' 'out', deinit tmp, etc
+      Symbol* p = convertParam(qt);
+      ret = new SymExpr(p);
     }
   }
 
-  // done generating actuals
-  exitCallActuals();
-
+  if (outRf) *outRf = rf;
 
+  return ret;
 }
 
-void TConverter::handlePostCallActions(CallExpr* c,
-                                       const Call* node,
-                                       const resolution::ResolvedExpression* re,
-                                       RV& rv) {
-}
+Expr* TConverter::convertNewCallOrNull(const Call* node, RV& rv) {
+  if (!node->calledExpression() ||
+      !node->calledExpression()->isNew()) {
+    return nullptr;
+  }
 
+  // We need the 'ResolvedExpression' to get the 'NEW_INIT' action.
+  auto re = rv.byAstOrNull(node);
+  if (!re) return nullptr;
 
-bool TConverter::enter(const Module* node, RV& rv) {
-  if (trace)
-    printf("enter module %s %s\n", node->id().str().c_str(), asttags::tagToString(node->tag()));
+  // Fetch the signature of the 'init()' procedure.
+  const TypedFnSignature* init = nullptr;
+  for (const auto& action : re->associatedActions()) {
+    if (action.action() == AssociatedAction::NEW_INIT) {
+      init = action.fn();
+      break;
+    }
+  }
 
-  if (modulesToConvert.count(node->id()) == 0) {
-    // this module should not be converted.
-    if (trace) printf("Not in modules to convert\n");
-    return false;
+  const ResolvedFunction* rf = nullptr;
+  if (paramElideCallOrNull(init, re->poiScope(), &rf)) {
+    INT_ASSERT(false && "Should not have param elided initializer call!");
+  } else if (!rf) {
+    return nullptr;
   }
 
-  if (node == symbol) {
-    // we are visiting it to convert the module initialization function
-    // so proceed with the traversal
-    if (trace) printf("Proceeding with module converting\n");
-    return true;
-  } else {
-    if (trace) printf("Recording a submodule\n");
-    submodulesEncountered.push_back(node);
+  // For 'new unmanaged C(...)' it should generate a call to a '_new'
```

Comment:
Is the `owned` part just not handled yet?

Reply:
Added a `TODO` about handling management, thanks!